oktaawscli/okta_auth.py

```diff
@@ -29,31 +32,46 @@ def __init__(self, okta_profile, verbose, logger,
         self.app = okta_auth_config.app_for(okta_profile)
         self.debug = debug
 
-        okta_info = os.path.join(os.path.expanduser('~'), '.okta-token')
+        okta_info = os.path.join(os.path.expanduser("~"), ".okta-token")
         if not os.path.isfile(okta_info):
-            open(okta_info, 'a').close()
+            open(okta_info, "a").close()
 
     def primary_auth(self):
-        """ Performs primary auth against Okta """
+        """Performs primary auth against Okta"""
         # check if token is valid
         session_id = self.get_cached_session_id()
         if session_id is not None:
             return session_id
         auth_data = {
             "username": self.okta_auth_config.username_for(self.okta_profile),
-            "password": self.okta_auth_config.password_for(self.okta_profile)
+            "password": self.okta_auth_config.password_for(self.okta_profile),
         }
-        resp = requests.post(self.https_base_url + '/api/v1/authn', json=auth_data)
+        resp = requests.post(self.https_base_url + "/api/v1/authn", json=auth_data)
         resp_json = resp.json()
-        if 'status' in resp_json:
-            if resp_json['status'] == 'MFA_REQUIRED':
-                factors_list = resp_json['_embedded']['factors']
-                state_token = resp_json['stateToken']
+        if "status" in resp_json:
+            if resp_json["status"] == "MFA_REQUIRED":
+                factors_list = resp_json["_embedded"]["factors"]
+                state_token = resp_json["stateToken"]
                 session_token = self.verify_mfa(factors_list, state_token)
-            elif resp_json['status'] == 'SUCCESS':
-                session_token = resp_json['sessionToken']
+            elif resp_json["status"] == "SUCCESS":
+                session_token = resp_json["sessionToken"]
+            elif resp_json["status"] == "MFA_ENROLL":
+                self.logger.warning(
+                    """MFA not enrolled. Cannot continue.
+                Please enroll an MFA factor in the Okta Web UI first!"""
+                )
+                sys.exit(2)
+            elif resp_json["status"] == "LOCKED_OUT":
+                self.logger.error(
+                    """Account is locked. Cannot continue.
+                Please contact you administrator in order to unlock the account!"""
+                )
+                sys.exit(1)
+            else:
+                self.logger.error("Unknown authentication status")
```

Comment:
i would add the value of `resp_json["status"]` to the exception message to give more context

Reply:
Done,